boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,162 @@
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream)
+            throws IOException, InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit(new Callable() {
+
+            public String call() throws IOException {
+
+                try {
+                    String line = null;
+
+                    while ((line = br.readLine()) != null) {
+                        //System.out.print(".");
+                        if (line.contains(READY_MARKER)) {
+                            break;
+                        }
+                    }
+                } finally {
+                    br.close();
+                }
+                return "IAM Microservice started";
+            }
+        });
+
+        // Check for timeout
+        try {
+            return future.get(TIMEOUT, TimeUnit.SECONDS);
+        } catch (TimeoutException toe) {
+            // Stop streaming the output, but the microservice continues to execute
+            br.close();
+            return "Warning: IAM Microservice timed out to start. See the logs for the details.";
+        } finally {
+            executor.shutdownNow();
+            future.cancel(true);
+        }
+    }
+
+    /**
+     * Prepare java command with jvm args
+     */
+    private static boolean buildJavaCommand(String paHome) {
+
+        String javaCmd = paHome + SEPARATOR + "jre" + SEPARATOR + "bin" + SEPARATOR + "java";
+
+        if (new File(javaCmd).exists()) {
+            command.add(javaCmd);
+            command.add("-jar");
+            command.addAll(Arrays.asList(JVM_ARGS));
+            return true;
+        } else {
+            throw new RuntimeException("Java command not found when starting IAM microservice");
+            //LOGGER.error("Java command not found when starting IAM microservice");
+            //System.exit(1);
+            //return false;
+        }
+
+    }
+
+    /**
+     * Check the microservice executable war
+     */
+    private static boolean buildMicroservicePath(String paHome, String microservicesPath) {
+
+        String microserviceFile = paHome + SEPARATOR + microservicesPath + SEPARATOR + MICROSERVICE_NAME;
+
+        if (new File(microserviceFile).exists()) {
+            command.add(microserviceFile);
+            return true;
+        } else {
+            throw new RuntimeException("IAM microservice is not deployed");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)